src/lib/PhpCsFixer/Config.php

```diff
@@ -0,0 +1,175 @@
+            'no_unneeded_final_method' => true,
+            'no_unused_imports' => true,
+            'no_whitespace_before_comma_in_array' => true,
+            'no_whitespace_in_blank_line' => true,
+            'normalize_index_brace' => true,
+            'object_operator_without_whitespace' => true,
+            'php_unit_fqcn_annotation' => true,
+            'phpdoc_annotation_without_dot' => true,
+            'phpdoc_indent' => true,
+            'phpdoc_inline_tag' => true,
+            'phpdoc_no_access' => true,
+            'phpdoc_no_alias_tag' => true,
+            'phpdoc_no_empty_return' => true,
+            'phpdoc_no_package' => true,
+            'phpdoc_no_useless_inheritdoc' => true,
+            'phpdoc_return_self_reference' => true,
+            'phpdoc_scalar' => true,
+            'phpdoc_separation' => true,
+            'phpdoc_single_line_var_spacing' => true,
+            'phpdoc_summary' => true,
+            'phpdoc_trim' => true,
+            'phpdoc_types' => true,
+            'phpdoc_types_order' => [
+                'null_adjustment' => 'always_last',
+                'sort_algorithm' => 'none',
+            ],
+            'phpdoc_var_without_name' => true,
+            'protected_to_private' => true,
+            'return_type_declaration' => true,
+            'semicolon_after_instruction' => true,
+            'short_scalar_cast' => true,
+            'single_class_element_per_statement' => true,
+            'single_line_comment_style' => [
+                'comment_types' => ['hash'],
+            ],
+            'single_quote' => true,
+            'single_trait_insert_per_statement' => true,
+            'standardize_increment' => true,
+            'standardize_not_equals' => true,
+            'ternary_operator_spaces' => true,
+            'trailing_comma_in_multiline_array' => true,
+            'trim_array_spaces' => true,
+            'unary_operator_spaces' => true,
+            'whitespace_after_comma_in_array' => true,
+            'dir_constant' => true,
+            'ereg_to_preg' => true,
+            'error_suppression' => true,
+            'fopen_flag_order' => true,
+            'fopen_flags' => ['b_mode' => false],
+            'function_to_constant' => true,
+            'implode_call' => true,
+            'is_null' => true,
+            'modernize_types_casting' => true,
+            'native_constant_invocation' => [
+                'fix_built_in' => false,
+                'include' => [
+                    'DIRECTORY_SEPARATOR',
+                    'PHP_SAPI',
+                    'PHP_VERSION_ID',
+                ],
+                'scope' => 'namespaced',
+            ],
+            'no_alias_functions' => true,
+            'no_homoglyph_names' => true,
+            'non_printable_character' => true,
+            'php_unit_construct' => true,
+            'php_unit_mock_short_will_return' => true,
+            'psr4' => true,
+            'self_accessor' => true,
```

Comment:
This would cause in .e.g. [eZ/Publish/API/Repository/Repository.php](https://github.com/ezsystems/ezpublish-kernel/blob/599e19a2b87a2f3316de436cf2aae6aab28e570f/eZ/Publish/API/Repository/Repository.php#L106)
``` diff
-    public function sudo(callable $callback, Repository $outerRepository = null);
+    public function sudo(callable $callback, self $outerRepository = null);
```
which we agreed is not what we want due to self/static issues in child classes.
// cc @Nattfarinn